R/locations_and_mascots.R

```diff
@@ -0,0 +1,233 @@
+## LOCATIONS AND MASCOTS (by: pep)
+## for: MLB, NBA, NCAA, NFL, and NHL (NOT: EPL or MLS)
+
+## please run the ncaa_color_update script
+## please run the logos.R script
+## make sure to load rvest after tidyverse (purr)
+library(tidyverse)
+library(readr)
+library(rvest) 
+
+
+## Removing epl and mls leagues
+
+teamcolors_without_epl_mls <- teamcolors %>%
+  filter(league == "mlb" | league == "nba" |
+           league == "nfl" | league == "nhl" |
+           league == "ncaa")
+
+## removing the weird X1 from the datasets
+teamcolors_without_epl_mls <- teamcolors_without_epl_mls %>%
+  select(-X1)
+
+
+## this is a function that is called on in the name_location_mascot_function below
+## the only thing that this function does is replaces all of the specificed strings to include an underscore
+## this allows the main function to seperate the name of the team into location and mascot, using pluck 
+
+nasty_gsub_function <- function(df){
+  
+  df <- mutate(df,    ## mlb, nfl, nba, nhl
+               
+               name = gsub("Golden State", "Golden_State", name),
+               name = gsub("Green Bay", "Green_Bay", name),
+               name = gsub("Kansas City", "Kansas_City", name),
+               name = gsub("Los Angeles", "Los_Angeles", name),
+               name = gsub("New England", "New_England", name),
+               name = gsub("New Jersey", "New_Jersey", name),
+               name = gsub("New Orleans", "New_Orleans", name),
+               name = gsub("New York", "New_York", name),
+               name = gsub("Oklahoma City", "Oklahoma_City", name),
+               name = gsub("San Antonio", "San_Antonio", name),
+               name = gsub("San Diego", "San_Diego", name),
+               name = gsub("San Francisco", "San_Francisco", name),
+               name = gsub("San Jose", "San_Jose", name),
+               name = gsub("St. Louis", "St._Louis", name),
+               name = gsub("Tampa Bay", "Tampa_Bay", name),
+               
+               ## ncaa (nasty part)
+               
+               name = gsub("Air Force", "Air_Force", name), 
+               name = gsub("Appalachian State", "Appalachian_State", name),
```

Comment:
We could get a little bit more clever here. What about something like `gsub(" State$", "_State", name)`?